specs/Extensions.md

```diff
@@ -0,0 +1,585 @@
+interface ICoreWebView2ProfileGetBrowserExtensionsCompletedHandler;
+
+/// Additional options used to create WebView2 Environment.
+[uuid(4B1F63E9-F7A5-4EA5-8D84-2B30F2404E82), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions6 : ICoreWebView2EnvironmentOptions5 {
+  /// When `AreBrowserExtensionsEnabled` is set to `TRUE`, new extensions can be added to user profile and used.
+  /// `AreBrowserExtensionsEnabled` is default to be `FALSE`, in this case, new extensions can't be installed, and
+  /// already installed extension won't be avaliable to use in user profile.
+  /// See `ICoreWebView2BrowserExtension` for Extensions API details.
+  [propget] HRESULT AreBrowserExtensionsEnabled([out, retval] BOOL* value);
+  /// Sets the `AreBrowserExtensionsEnabled` property.
+  [propput] HRESULT AreBrowserExtensionsEnabled([in] BOOL value);
+}
+
+/// This is the ICoreWebView2 profile.
+[uuid(8B16D238-9508-4C36-B4D4-749EB9AC4AD0), object, pointer_default(unique)]
+interface ICoreWebView2Profile6 : IUnknown {
+    /// Adds the [browser extension](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions) using the extension path for unpacked extensions
+    /// from the local device. The extension folder path is the topmost folder of an unpacked browser extension and contains the browser extension manifest file.
+    /// If the `extensionFolderPath` is an invalid path or doesn't contain the extension manifest.json file, this function will return E_FAIL to callers. 
+    /// Installed extension will default `IsEnabled` to true.
+    /// When `AreBrowserExtensionsEnabled` is `FALSE`, `AddBrowserExtension` will fail and return HRESULT `ERROR_NOT_SUPPORTED`.
+    /// During installation, the content of the extension is not copied to the user data folder. Once the extension is installed, changing the 
+    /// content of the extension will cause the extension to be removed from the installed profile. 
+    /// When an extension is added the extension is persisted in the corresponding profile. The extension will still be installed the next time you use this profile.
+    /// The following summarizes the possible error values that can be returned from 
+    /// `AddBrowserExtension` and a description of why these errors occur.
+    ///
+    /// Error value                                     | Description
+    /// ----------------------------------------------- | --------------------------
+    /// `HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED)`       | Extensions are disabled due to policy rules.
+    /// `HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)`      | Cannot find `manfiest.json` file or it is not a valid extension manfiest.
+    /// `E_ACCESSDENIED`                                | Cannot load extension with file or directory name starting with \"_\", reserved for use by the system.
+    /// `E_FAIL`                                        | Extension failed to install with other unknown reasons. 
```

Comment:
It sounds like it's not possible to have >1 extension in the same directory, since we only claim to support directories that are the result of "an unpacked browser extension".  Is that right?  If there are multiple manifest.json in the provided directory for any reason is that an E_FAIL case?